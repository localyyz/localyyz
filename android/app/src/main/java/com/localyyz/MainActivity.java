package com.localyyz;

import android.content.Intent;
import android.os.Bundle;

import com.facebook.react.ReactActivity;
import com.facebook.react.ReactActivityDelegate;

import javax.annotation.Nullable;

import io.branch.rnbranch.RNBranchModule;

public class MainActivity extends ReactActivity {

    /**
     * Returns the name of the main component registered from JavaScript.
     * This is used to schedule rendering of the component.
     */
    @Override
    protected String getMainComponentName() {
        return "localyyz";
    }

    @Override
    protected void onStart() {
        super.onStart();
        RNBranchModule.initSession(getIntent().getData(), this);
    }

    @Override
    public void onNewIntent(Intent intent) {
        setIntent(intent);
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        MainApplication.getCallbackManager().onActivityResult(requestCode, resultCode, data);
    }

    @Override
    protected ReactActivityDelegate createReactActivityDelegate() {
        return new ReactActivityDelegate(this, getMainComponentName()) {
            @Nullable
            @Override
            protected Bundle getLaunchOptions() {
                Bundle initialProps = new Bundle();
                initialProps.putString("API_URL", BuildConfig.API_URL);
                return initialProps;
            }
        };
    }

}
