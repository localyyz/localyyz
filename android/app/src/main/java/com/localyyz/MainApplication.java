package com.localyyz;

import android.app.Application;

import com.BV.LinearGradient.LinearGradientPackage;
import com.cmcewen.blurview.BlurViewPackage;
import com.facebook.CallbackManager;
import com.facebook.FacebookSdk;
import com.facebook.appevents.AppEventsLogger;
import com.facebook.react.ReactApplication;
import com.geektime.rnonesignalandroid.ReactNativeOneSignalPackage;
import com.idehub.GoogleAnalyticsBridge.GoogleAnalyticsBridgePackage;
import com.brentvatne.react.ReactVideoPackage;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.shell.MainReactPackage;
import com.facebook.reactnative.androidsdk.FBSDKPackage;
import com.facebook.soloader.SoLoader;
import com.learnium.RNDeviceInfo.RNDeviceInfo;
import com.microsoft.codepush.react.CodePush;
import com.oblador.vectoricons.VectorIconsPackage;
import com.reactnative.photoview.PhotoViewPackage;

import java.util.Arrays;
import java.util.List;

import ca.jaysoo.extradimensions.ExtraDimensionsPackage;
import io.branch.referral.Branch;
import io.branch.rnbranch.RNBranchPackage;

//import com.reactnativepayments.ReactNativePaymentsPackage;

public class MainApplication extends Application implements ReactApplication {

  private static CallbackManager mCallbackManager = CallbackManager.Factory.create();

  protected static CallbackManager getCallbackManager() {
    return mCallbackManager;
  }

  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {

    @Override
    protected String getJSBundleFile() {
      return CodePush.getJSBundleFile();
    }
    
    @Override
    public boolean getUseDeveloperSupport() {
      return BuildConfig.DEBUG;
    }

    @Override
    protected List<ReactPackage> getPackages() {
      return Arrays.<ReactPackage>asList(
          new MainReactPackage(),
            new ReactNativeOneSignalPackage(),
            new GoogleAnalyticsBridgePackage(),
            new ReactVideoPackage(),
            new ExtraDimensionsPackage(),
              new CodePush(BuildConfig.CODEPUSH_KEY, getApplicationContext(), BuildConfig.DEBUG),
              new RNBranchPackage(),
              //new ReactNativePaymentsPackage(),
              new FBSDKPackage(mCallbackManager),
              new PhotoViewPackage(),
              new VectorIconsPackage(),
              new LinearGradientPackage(),
              new RNDeviceInfo(),
              new BlurViewPackage()
      );
    }

    @Override
    public List<ReactPackage> createAdditionalReactPackages() {
      return getPackages();
    }


    @Override
    protected String getJSMainModuleName() {
      return "index";
    }
  };

  @Override
  public ReactNativeHost getReactNativeHost() {
    return mReactNativeHost;
  }

  @Override
  public void onCreate() {
    super.onCreate();

    FacebookSdk.sdkInitialize(getApplicationContext());
    // If you want to use AppEventsLogger to log events.
    AppEventsLogger.activateApp(this);

    // Branch
    Branch.getAutoInstance(this);

    SoLoader.init(this, /* native exopackage */ false);
  }

}
