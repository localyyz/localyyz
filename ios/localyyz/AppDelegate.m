/**
 * Copyright (c) 2015-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import "AppDelegate.h"

//#import <React/RCTBundleURLProvider.h>
#import <React/RCTRootView.h>

// Branch
#import <react-native-branch/RNBranch.h>

// Facebook
#import <FBSDKCoreKit/FBSDKCoreKit.h>
#import <FBSDKLoginKit/FBSDKLoginKit.h>
#import <FBNotifications/FBNotifications.h>
#import <UserNotifications/UserNotifications.h>

// CodePush
#import <CodePush/CodePush.h>

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  NSURL *jsCodeLocation;

  if (RCT_DEBUG == 1) {
    //jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
    jsCodeLocation = [NSURL URLWithString:@"http://localhost:8081/index.bundle?platform=ios&dev=true"];
  } else {
    jsCodeLocation = [CodePush bundleURL];
  }

  NSString *apiUrl = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"API_URL"];
  NSDictionary *props = @{@"API_URL" : apiUrl};
  
  RCTRootView *rootView = [[RCTRootView alloc] initWithBundleURL:jsCodeLocation
                                                      moduleName:@"localyyz"
                                               initialProperties:props
                                                   launchOptions:launchOptions];
  rootView.backgroundColor = [[UIColor alloc] initWithRed:1.0f green:1.0f blue:1.0f alpha:1];
  
  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  UIViewController *rootViewController = [UIViewController new];
  rootViewController.view = rootView;
  self.window.rootViewController = rootViewController;
  [self.window makeKeyAndVisible];

  // replace RN load view with the launch screen
  UIView* launchScreenView = [[[NSBundle mainBundle] loadNibNamed:@"LaunchScreen" owner:self options:nil] objectAtIndex:0];
  launchScreenView.frame = self.window.bounds;
  rootView.loadingView = launchScreenView;

  // facebook
  [[FBSDKApplicationDelegate sharedInstance] application:application didFinishLaunchingWithOptions:launchOptions];

  // branch
  [RNBranch initSessionWithLaunchOptions:launchOptions isReferrable:YES];

  // push notification
  UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
  center.delegate = self;
  [center requestAuthorizationWithOptions:(UNAuthorizationOptionSound | UNAuthorizationOptionAlert | UNAuthorizationOptionBadge) completionHandler:^(BOOL granted, NSError * _Nullable error) {
    if( !error ) {
      // required to get the app to do anything at all about push notifications
      [[UIApplication sharedApplication] registerForRemoteNotifications];
      NSLog( @"Push registration success.");
    } else {
      NSLog( @"Push registration FAILED" );
      NSLog( @"ERROR: %@ - %@", error.localizedFailureReason, error.localizedDescription );
      NSLog( @"SUGGESTIONS: %@ - %@", error.localizedRecoveryOptions, error.localizedRecoverySuggestion );
    }
  }];

  return YES;
}

- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo fetchCompletionHandler:(nonnull void (^)(UIBackgroundFetchResult))completionHandler {
  FBNotificationsManager *notificationsManager = [FBNotificationsManager sharedManager];
  [notificationsManager presentPushCardForRemoteNotificationPayload:userInfo
                                                 fromViewController:nil
                                                         completion:^(FBNCardViewController * _Nullable viewController, NSError * _Nullable error) {
                                                           if (error) {
                                                             completionHandler(UIBackgroundFetchResultFailed);
                                                           } else {
                                                             completionHandler(UIBackgroundFetchResultNewData);
                                                           }
                                                         }];
}

- (BOOL)application:(UIApplication *)application
            openURL:(NSURL *)url
            options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options {
  
  BOOL handled = [[FBSDKApplicationDelegate sharedInstance] application:application
                                                                openURL:url
                                                      sourceApplication:options[UIApplicationOpenURLOptionsSourceApplicationKey]
                                                             annotation:options[UIApplicationOpenURLOptionsAnnotationKey]
                  ];
  // Add any custom logic here.
  return handled;
}

// track opening and resuming the application attached to the push notification
- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo {
  [FBSDKAppEvents logPushNotificationOpen:userInfo];
}

// track action performed that was attached to the push notification
- (void)application:(UIApplication *)application handleActionWithIdentifier:(NSString *)identifier forRemoteNotification:(NSDictionary *)userInfo
  completionHandler:(void (^)())completionHandler {
  [FBSDKAppEvents logPushNotificationOpen:userInfo action:identifier];
}

- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity restorationHandler:(void (^)(NSArray *restorableObjects))restorationHandler {
  return [RNBranch continueUserActivity:userActivity];
}

- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
  NSString *token = [[deviceToken description] stringByTrimmingCharactersInSet: [NSCharacterSet characterSetWithCharactersInString:@"<>"]];
  token = [token stringByReplacingOccurrencesOfString:@" " withString:@""];
  NSLog(@"content--- %@", token);
  [FBSDKAppEvents setPushNotificationsDeviceToken:deviceToken];
}

// Facebook SDK
- (void)applicationDidBecomeActive:(UIApplication *)application {
  [FBSDKAppEvents activateApp];
}

@end
