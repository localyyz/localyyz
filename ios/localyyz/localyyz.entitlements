<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>aps-environment</key>
	<string>development</string>
	<key>com.apple.developer.associated-domains</key>
	<array>
		<string>applinks:localyyz.app.link</string>
		<string>applinks:localyyz-alternate.app.link</string>
		<string>applinks:localyyz.test-app.link</string>
		<string>applinks:localyyz-alternate.test-app.link</string>
	</array>
	<key>com.apple.developer.in-app-payments</key>
	<array>
		<string>merchant.com.localyyz.toronto</string>
	</array>
</dict>
</plist>
