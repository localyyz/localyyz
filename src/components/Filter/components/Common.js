import React from "react";
import {
  TouchableOpacity,
  StyleSheet,
  View,
  Text,
  FlatList
} from "react-native";

// third party
import PropTypes from "prop-types";

// custom
import { Styles, Sizes } from "localyyz/constants";
import { capitalize } from "localyyz/helpers";
import { GA } from "localyyz/global";

// local
import ExpandableHeader from "./ExpandableHeader";
import SelectedFilter from "./SelectedFilter";

export default class Common extends React.Component {
  static propTypes = {
    asyncFetch: PropTypes.func.isRequired,
    setFilter: PropTypes.func.isRequired,
    clearFilter: PropTypes.func.isRequired,
    data: PropTypes.object,
    selected: PropTypes.string,
    title: PropTypes.string
  };

  static defaultProps = {
    data: [],
    selected: "",
    title: ""
  };

  constructor(props) {
    super(props);
    this.state = { collapsed: true };
  }

  onToggle = () => {
    GA.trackEvent("filter/sort", "filter by " + this.props.title);
    this.setState({ collapsed: !this.state.collapsed });
  };

  onSelect = item => {
    this.props.title == "Brand"
      ? GA.trackEvent(
          "filter/sort",
          "filter by " + this.props.title,
          this.props.item
        )
      : "";
    this.setState({ collapsed: true });
    this.props.setFilter(item);
  };

  componentDidUpdate(_, prevState) {
    // TODO: have to track prevState or mobx update + did update
    // will be on a loop
    if (prevState.collapsed && !this.state.collapsed) {
      this.props.asyncFetch();
    }
  }

  // TODO: render sign this is selected
  renderItem = ({ item }) => {
    return (
      <TouchableOpacity onPress={() => this.onSelect(item)}>
        <View style={styles.container}>
          <Text style={styles.label}>{capitalize(item)}</Text>
        </View>
      </TouchableOpacity>
    );
  };

  render() {
    return (
      <View style={styles.container}>
        <TouchableOpacity onPress={this.onToggle}>
          <ExpandableHeader isOpen={!this.state.collapsed}>
            {this.props.title}
          </ExpandableHeader>
        </TouchableOpacity>
        {!this.state.collapsed || this.props.selected ? (
          <View style={styles.content}>
            {this.props.selected ? (
              <View style={styles.selected}>
                <SelectedFilter
                  ref={`${this.props.title}Filter`}
                  onClear={this.props.clearFilter}>
                  {this.props.selected}
                </SelectedFilter>
              </View>
            ) : (
              <FlatList
                scrollEnabled={false}
                data={this.props.data}
                keyExtractor={item => item}
                renderItem={this.renderItem}/>
            )}
          </View>
        ) : null}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    marginVertical: Sizes.InnerFrame / 2
  },

  content: {
    paddingVertical: Sizes.InnerFrame / 2
  },

  label: {
    ...Styles.Text,
    ...Styles.EmphasizedText
  },

  selected: {
    alignItems: "flex-start",
    marginVertical: Sizes.InnerFrame / 2
  }
});
