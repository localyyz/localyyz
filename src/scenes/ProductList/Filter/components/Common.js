import React from "react";
import { TouchableOpacity, StyleSheet, View, Text } from "react-native";

// third party
import PropTypes from "prop-types";
import { withNavigation } from "react-navigation";
import { observer, inject } from "mobx-react/native";
import MaterialIcon from "react-native-vector-icons/MaterialIcons";

// custom
import { Styles, Sizes, Colours } from "localyyz/constants";
import { capitalize } from "localyyz/helpers";
import { GA } from "localyyz/global";

@inject(stores => ({
  filterStore: stores.filterStore
}))
@observer
export class Common extends React.Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    asyncFetch: PropTypes.func.isRequired,
    setFilter: PropTypes.func.isRequired,
    clearFilter: PropTypes.func.isRequired,

    // optional
    //
    // scene name
    scene: PropTypes.string,
    selected: PropTypes.string,
    title: PropTypes.string,
    icon: PropTypes.string,
    onPress: PropTypes.func,
    isButton: PropTypes.bool // render as inline button
  };

  static defaultProps = {
    data: [],
    title: "",
    isButton: false,
    icon: null
  };

  constructor(props) {
    super(props);

    // bindings
    this.onSelect = this.onSelect.bind(this);
  }

  onSelect() {
    GA.trackEvent("filter/sort", "filter by " + this.props.title);
    this.props.onPress
      ? this.props.onPress()
      : this.props.navigation.push(this.props.scene || "FilterList", {
          ...this.props,
          title: this.props.title || this.props.label,
          filterStore: this.props.filterStore
        });
  }

  get renderSelectedOption() {
    return (
      <View style={[styles.optionContent, styles.selectedOption]}>
        <Text
          style={[styles.optionLabel, styles.selectedOptionLabel]}
          numberOfLines={1}>
          {capitalize(this.props.selected)}
        </Text>
      </View>
    );
  }

  render() {
    return this.props.isButton ? (
      <TouchableOpacity
        style={{ flex: 1 }}
        onPress={this.onSelect}
        hitSlop={{
          top: Sizes.InnerFrame * 2,
          bottom: Sizes.InnerFrame * 2,
          left: Sizes.InnerFrame,
          right: Sizes.InnerFrame
        }}>
        <View style={[styles.option]}>
          <Text
            style={[styles.optionLabel]}
            testID={`filter${this.props.id}`}
            numberOfLines={1}>
            {this.props.selected
              ? this.renderSelectedOption
              : capitalize(this.props.title)}
          </Text>
          {this.props.icon ? (
            <MaterialIcon
              name="playlist-add"
              size={Sizes.Text}
              color={Colours.Text}/>
          ) : (
            <View />
          )}
        </View>
      </TouchableOpacity>
    ) : (
      <TouchableOpacity onPress={this.onSelect}>
        <View style={styles.container}>
          <Text style={styles.headerLabel}>{capitalize(this.props.title)}</Text>
          <View style={styles.selectedValue}>
            <Text style={styles.selectedValueLabel}>
              {capitalize(this.props.selected || "All")}
            </Text>
          </View>
        </View>
      </TouchableOpacity>
    );
  }
}

export default withNavigation(Common);

const styles = StyleSheet.create({
  container: {
    ...Styles.Horizontal,
    ...Styles.EqualColumns,
    marginVertical: Sizes.InnerFrame
  },

  option: {
    ...Styles.Horizontal,

    borderRadius: Sizes.RoundedBorders,
    marginTop: 1,
    justifyContent: "space-around",

    flex: 1,
    height: Sizes.Button,
    marginHorizontal: Sizes.InnerFrame / 4,
    paddingHorizontal: Sizes.InnerFrame / 4,
    backgroundColor: Colours.Action
  },

  optionContent: {
    ...Styles.Horizontal,
    justifyContent: "center",

    width: Sizes.Width / 4,
    height: Sizes.Button
  },

  optionLabel: {
    ...Styles.SmallText,
    ...Styles.Emphasized
  },

  selectedOption: {
    ...Styles.Horizontal,
    justifyContent: "center",

    width: Sizes.Width / 4,
    height: Sizes.Button
  },

  selectedOptionLabel: {
    ...Styles.SmallText,
    ...Styles.Emphasized
  },

  headerLabel: {
    ...Styles.Text,
    ...Styles.Emphasized,
    fontSize: Sizes.H2
  },

  selectedValueLabel: {
    ...Styles.Text
  }
});
