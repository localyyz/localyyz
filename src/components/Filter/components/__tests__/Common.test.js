import React from "react";

// custom
import { renderWithLayout } from "localyyz/tests";

// local
import { Common } from "../Common";

// constants
const NAME = "Common";
const PROPS = {
  asyncFetch: jest.fn(),
  setFilter: jest.fn(),
  clearFilter: jest.fn(),
  data: jest.mock(),
  selected: "",
  title: "filter1"
};

describe(NAME, () => {
  it("should render properly", () => {
    let c = renderWithLayout(<Common {...PROPS} />).getInstance();
    expect(c).toBeDefined();
  });

  it("should render to snapshot", () => {
    expect(renderWithLayout(<Common {...PROPS} />).toJSON()).toMatchSnapshot();
  });
});
