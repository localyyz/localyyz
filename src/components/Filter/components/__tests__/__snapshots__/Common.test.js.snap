// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common should render expanded properly 1`] = `
Array [
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <ExpandableHeader
      isOpen={true}
    >
      filter1
    </ExpandableHeader>
  </TouchableOpacity>,
  <Component
    style={
      Object {
        "paddingVertical": 7.5,
      }
    }
  >
    <FlatList
      data={
        Object {
          "addMatchers": [Function],
          "advanceTimersByTime": [Function],
          "autoMockOff": [Function],
          "autoMockOn": [Function],
          "clearAllMocks": [Function],
          "clearAllTimers": [Function],
          "deepUnmock": [Function],
          "disableAutomock": [Function],
          "doMock": [Function],
          "dontMock": [Function],
          "enableAutomock": [Function],
          "fn": [Function],
          "genMockFromModule": [Function],
          "isMockFunction": [Function],
          "mock": [Function],
          "requireActual": [Function],
          "requireMock": [Function],
          "resetAllMocks": [Function],
          "resetModuleRegistry": [Function],
          "resetModules": [Function],
          "restoreAllMocks": [Function],
          "runAllImmediates": [Function],
          "runAllTicks": [Function],
          "runAllTimers": [Function],
          "runOnlyPendingTimers": [Function],
          "runTimersToTime": [Function],
          "setMock": [Function],
          "setTimeout": [Function],
          "spyOn": [Function],
          "unmock": [Function],
          "useFakeTimers": [Function],
          "useRealTimers": [Function],
        }
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      renderItem={[Function]}
      scrollEnabled={false}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </Component>,
]
`;

exports[`Common should render item properly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Component
    style={
      Object {
        "marginVertical": 7.5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "color": "rgb(30, 30, 30)",
          "fontSize": 16,
          "fontWeight": "300",
        }
      }
    >
      Some Filter
    </Text>
  </Component>
</TouchableOpacity>
`;

exports[`Common should render to snapshot 1`] = `
<View
  style={
    Object {
      "marginVertical": 7.5,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "marginRight": 7.5,
          }
        }
      >
        filter1
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(30, 30, 30)",
              "fontSize": 16,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        îŒ“
      </Text>
    </View>
  </View>
</View>
`;
